kolibri/core/assets/src/components.js

```diff
@@ -95,6 +95,89 @@ var Tag = Mn.ItemView.extend({
     }
 });
 
+
+// Implements CRUD actions for a given item
+var CrudItem = Mn.ItemView.extend({
```

Comment:
I think this type of abstraction pattern works well to reduce code duplication when all types of "items" are basic key/value pair lists. This is particularly true when there are many item types with only the names of their attributes changing between types.

However this kind of abstraction layer can make a lot of basic UI customizations more difficult:

* change the order in which attributes are displayed
* add additional grouping or formatting to particular sets of attributes
* handle new types of inputs: boolean, multi-select, text area...
* incorporate validation

If these sorts of customization become necessary on a per-item-type basis, it can tip the balance and the extra abstraction layer becomes more complex than just writing a new view customized for each type of item.